flash_attn/cute/flash_fwd.py

```diff
@@ -1526,16 +1581,44 @@ def load(
                     with cute.arch.elect_one():
                         cute.arch.mbarrier_arrive_and_expect_tx(mbar_ptr_Q, self.tma_copy_q_bytes)
                     load_Q(tma_bar_ptr=mbar_ptr_Q)
-                n_block_min, n_block_max = block_info.get_n_block_min_max(seqlen, m_block)
-                # if cute.arch.thread_idx()[0] == 0:
-                #     cute.printf("m_block = %d, n_block_min: %d, n_block_max: %d", m_block, n_block_min, n_block_max)
-                for i in cutlass.range(n_block_max - n_block_min, unroll=2):
-                    n_block = n_block_max - i - 1
-                    pipeline_k.producer_acquire(kv_producer_state)
-                    load_K(src_idx=n_block, producer_state=kv_producer_state)
-                    pipeline_v.producer_acquire(kv_producer_state)
-                    load_V(src_idx=n_block, producer_state=kv_producer_state)
-                    kv_producer_state.advance()
+
```

Comment:
I would put the non-blocksparse case first (that's the simpler case to read) before the blocksparse case